neqo-http3/src/connection.rs

```diff
@@ -796,22 +799,60 @@ impl Http3Connection {
             .as_request_target()
             .map_err(|_| Error::InvalidRequestTarget)?;
 
-        // Transform pseudo-header fields
-        let mut final_headers = vec![
-            Header::new(":method", request.method),
-            Header::new(":scheme", target.scheme()),
-            Header::new(":authority", target.authority()),
-            Header::new(":path", target.path()),
-        ];
-        if let Some(conn_type) = request.connect_type {
-            final_headers.push(Header::new(":protocol", conn_type.string()));
+        match request.connect_type {
+            Some(_) if request.method != "CONNECT" => {
+                qwarn!(
+                    "Method {} with CONNECT type {:?}",
+                    request.method,
+                    request.connect_type
+                );
+                debug_assert!(false);
```

Comment:
Is this really the way we want to handle this?  I see the test (which will fail in a non-debug test run...) but it seems like the right thing to do is abort: the request we're talking about isn't something that makes sense.

It's understandable that you want to pass a request through at this point, but you aren't validating earlier in the stack either.

Reply:
Addressed in https://github.com/mozilla/neqo/pull/2886/commits/577307073ff708e4db240c2609527a94660045a7. Let me know if I misunderstood your suggestion.